site/docs/contribute/modules.md

```diff
@@ -2,20 +2,50 @@
 1. Create and publish the [Module Helm Chart](#module-helm-chart) that will be used by the control plane to deploy the module workload, update it, and delete it as necessary.
 1. Create the [FybrikModule YAML](#fybrikmodule-yaml) which describes the capabilities of the module workload, in which flows it should be considered for inclusion, its supported interfaces, and the link to the module helm chart.
-1. [Test](#test) the new module
+1. [Test](#test) the new module.
+
+These steps are described in the following sections in more detail, so that you can create your own modules for use by Fybrik.  Note that a new module is maintained in its own git repository, separate from the [Fybrik](https://github.com/fybrik/fybrik) repository.
 
-These steps are described in the following sections in more detail, so that you can create your own modules for use by Fybrik.  Note that a new module is maintained in its own git repository, separate from the [fybrik](https://github.com/fybrik/fybrik) repository.
+before you start, consider using the [template module](https://github.com/fybrik/template-module), where you can create a new module repository with all the needed templates for the files you need.
 
 ## Module Workload
 
-The module workload is associated with a specific user workload and is deployed by the control plane.  It may implement the logic required itself, or it may be a client interface to an external component.  The former will have module type "server" and the latter "config".
+The module workload is associated with a specific user workload and is deployed by the control plane.  
+It may implement the logic required itself, or it may be a client interface to an external component.  
+The former will have module type "server" and the latter "config".
+
+There is also a third type of module workload known as a plugin.  
+It provides a standard interface by which another module may invoke its capabilities.  
+For example, you may have a module that reads data but doesn't know how to do data transforms.  
+Rather than implementing transforms in the module workload code, it can call the plugin to do the transforms.  
+The control plane deploys the relevant transform plugin as well as the read module.
+
+### Configuration
+
+Modules receive the parameters that define the configuration needed for the module (such as data asset connection information, required transformations, and so on) as Helm chart values.  
+The list of the parameters passed, and an example of parameters passed, can be found in the [Helm values passed to the module](#helm-values-passed-to-the-module) section.
+
+To read the parameters, most modules define a conf.yaml file that grabs the relevant Helm values, and is copied to the environment of the module container.
+
+An example of a conf.yaml of the AirByte module can be found [here](https://github.com/fybrik/airbyte-module/blob/main/helm/abm/files/conf.yaml). 
```

Comment:
You should put links to a specific version branch to take the `conf.yaml` file from. Because it is changing in the `main` branch.

Reply:
Done